### PR TITLE
fix typos

### PR DESCRIPTION
**Pull request by demoray**
This mostly fixes comments, but in a handful of places fixes bugs due to typos.

One example:
```
-            insn_op_idx=None if cmsg.operand_idx == -1 else cmsg.opearnd_idx,
+            insn_op_idx=None if cmsg.operand_idx == -1 else cmsg.operand_idx,
```

### CODE REVIEW

File: angr/analyses/decompiler/optimization_passes/duplication_reverter/duplication_reverter.py [hunk 1]
```diff
@@ -1190,7 +1190,7 @@ def shared_common_conditional_dom(self, nodes, graph: nx.DiGraph):
         node_level = [node]
         seen_nodes = set()
         while node_level:
-            # check if any of the nodes on the current level are dominaters to all nodes
```
Comment (twizmwazin) [inline angr/analyses/decompiler/optimization_passes/duplication_reverter/duplication_reverter.py:1193]: I think this was correct?

File: angr/knowledge_plugins/key_definitions/atoms.py [hunk 2]
```diff
@@ -67,7 +67,7 @@ def from_argument(
         argument: SimFunctionArgument, arch: Arch, full_reg=False, sp: int | None = None
     ) -> Register | MemoryLocation:
         """
-        Instanciate an `Atom` from a given argument.
+        instantiate an `Atom` from a given argument.
```
Comment (twizmwazin) [inline angr/knowledge_plugins/key_definitions/atoms.py:70]: I think this should be capitalized, possibly add an s at the end?

File: angr/procedures/posix/strtok_r.py [hunk 3]
```diff
@@ -26,8 +26,8 @@ def run(self, str_ptr, delim_ptr, save_ptr, str_strlen=None, delim_strlen=None):
         strstr = angr.SIM_PROCEDURES["libc"]["strstr"]
         strlen = angr.SIM_PROCEDURES["libc"]["strlen"]
 
-        l.debug("Doin' a strtok_r!")
```
Comment (twizmwazin) [inline angr/procedures/posix/strtok_r.py:29]: I think this one was intentional? If adding a g we should remove the '

File: angr/analyses/decompiler/optimization_passes/duplication_reverter/duplication_reverter.py (hunk 1 above)
Comment (ltfish) [reply angr/analyses/decompiler/optimization_passes/duplication_reverter/duplication_reverter.py:1193]: It should have been "dominators".

File: angr/analyses/decompiler/region_simplifiers/expr_folding.py [hunk 4]
```diff
@@ -163,7 +163,7 @@ class ExpressionUseFinder(AILBlockWalker):
 
         v16 = ((int)v23->field_5) + 1 & 255;
         v23->field_5 = ((char)(((int)v23->field_5) + 1 & 255));
-        v13 = printf("Recieved packet %d for connection with %d\\n", v16, a0 & 255);
+        v13 = printf("Received packet %d for connection with %d\\n", v16, a0 & 255);
```
Comment (ltfish) [inline angr/analyses/decompiler/region_simplifiers/expr_folding.py:166]: It was misspelled as "Recieved" in the original binary.

File: angr/procedures/posix/strtok_r.py (hunk 3 above)
Comment (ltfish) [reply angr/procedures/posix/strtok_r.py:29]: This one was intentional.

File: angr/analyses/decompiler/region_simplifiers/expr_folding.py (hunk 4 above)
Comment (demoray) [reply angr/analyses/decompiler/region_simplifiers/expr_folding.py]: undone.

File: angr/procedures/posix/strtok_r.py (hunk 3 above)
Comment (demoray) [reply angr/procedures/posix/strtok_r.py:29]: undone

File: angr/knowledge_plugins/key_definitions/atoms.py (hunk 2 above)
Comment (demoray) [reply angr/knowledge_plugins/key_definitions/atoms.py]: I think it should be capitalized, but this aligns with the other functions in the file:

```
        Create a Register atom.
        Create a MemoryLocation atom,
```

Comment (demoray) [reply angr/knowledge_plugins/key_definitions/atoms.py]: updated

File: angr/analyses/decompiler/optimization_passes/duplication_reverter/duplication_reverter.py (hunk 1 above)
Comment (demoray) [reply angr/analyses/decompiler/optimization_passes/duplication_reverter/duplication_reverter.py:1193]: fixed.

**Pull request merged**